### ISSUE TITLE
Unable to create Ctable using bcolz

### ISSUE THREAD

**Issue by supriyom083** [Blosc/bcolz#252]
with bcolz.ctable(np.empty(0, dtype="i4,f8"),rootdir='mydir', mode="w") as ct:
    for i in xrange(N):
        ct.append((i, i**2))

AttributeError                            Traceback (most recent call last)
<ipython-input-29-85dc4adb8e77> in <module>()
----> 1 with bcolz.ctable(np.empty(0, dtype="i4,f8"),rootdir='mydir', mode="w") as ct:
      2     for i in xrange(N):
      3         ct.append((i, i**2))

AttributeError: __exit__

**Comment by supriyom083** [comment]
bcolz version '0.9.0'

**Comment by FrancescAlted** [comment]
I cannot reproduce this error with:

```
import numpy as np
import bcolz

N = 10

with bcolz.ctable(np.empty(0, dtype="i4,f8"), rootdir='mydir', mode="w") as ct:
    for i in range(N):
        ct.append((i, i**2))

with bcolz.ctable(rootdir='mydir', mode="r") as ct2:
    print("ct:", ct2)
```

that triggered this error:

```
Traceback (most recent call last):
  File "prova.py", line 11, in <module>
    print("ct:", ct2)
  File "/Users/faltet/software/bcolz/bcolz/ctable.py", line 1267, in __exit__
    self.flush()
  File "/Users/faltet/software/bcolz/bcolz/ctable.py", line 1213, in flush
    self.cols[name].flush()
  File "bcolz/carray_ext.pyx", line 2585, in bcolz.carray_ext.carray.flush (/Users/faltet/software/bcolz/bcolz/carray_ext.c:31509)
    self.chunks.flush(chunk_)
  File "bcolz/carray_ext.pyx", line 803, in bcolz.carray_ext.chunks.flush (/Users/faltet/software/bcolz/bcolz/carray_ext.c:10847)
    self._save(self.nchunks, chunk_)
  File "bcolz/carray_ext.pyx", line 788, in bcolz.carray_ext.chunks._save (/Users/faltet/software/bcolz/bcolz/carray_ext.c:10485)
    raise IOError(
OSError: cannot modify data because mode is 'r'
```

which should be fixed by 31a4fa8.  Does that fixes your problem?

Also, from your description, it looks like you are using a Python version that does not support contexts.  Please bear in mind that you need Python 2.6 or later to run bcolz.

**Comment by FrancescAlted** [comment]
Closing this.  Please feel free to reopen in case you continue having problems.
